### PR TITLE
Multi query pagination

### PR DESCRIPTION
**Pull request by aadilhasan**

### CODE REVIEW

File: src/lib/apis/fetchMultipleQueries.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
+import { fetchGql } from "../utils/fetcher";
+import {
+  Config,
+  FetchQueryReturnType,
+  ResponseType,
+  Variables,
+} from "../types";
+import { chacheResponse, getFromCache } from "../cache";
```
Comment (sharathkrml) [inline src/lib/apis/fetchMultipleQueries.ts:8]:
```suggestion
import { cacheResponse, getFromCache } from "../cache";
```

File: src/lib/utils/removeQueriesIfNoNextPage.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+  // remove unsed variables
+  variablesToDelete = variablesToDelete.filter((deletedVariable) => {
+    const { assignedVariable } = deletedVariable;
+    return !remainingVariables.find((remainingVariable) => {
+      return remainingVariable.assignedVariable === assignedVariable;
+    });
+  });
+
+  const defination = queryDocument.definitions[0] as OperationDefinitionNode;
```
Comment (sharathkrml) [inline src/lib/utils/removeQueriesIfNoNextPage.ts:68]:
```suggestion
  const definition = queryDocument.definitions[0] as OperationDefinitionNode;
```

File: src/lib/utils/removeQueriesIfNoNextPage.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    if (queriesWithLastPage[queryName]) {
+      variablesToDelete = [...variablesToDelete, ...queryVariables];
+      return false;
+    }
+    remainingVariables = [...remainingVariables, ...queryVariables];
+    return true;
+  });
+
+  // no qery was deleted so return null
```
Comment (sharathkrml) [inline src/lib/utils/removeQueriesIfNoNextPage.ts:57]:
```suggestion
  // no query was deleted so return null
```

File: src/lib/apis/fetchQuery.ts [hunk 4]
```diff
@@ -14,71 +16,76 @@ const defaultConfig: Config = {
...
-          cursor: pageInfo?.nextCursor,
-        },
-        config
-      );
-    }
-    return null;
-  };
+    const handleNext = async () => {
+      if (hasNextPage) {
+        prevCursorsCache.push(prevCursors);
```
Comment (aadilhasan) [inline src/lib/apis/fetchQuery.ts:51]: why these changes are here ?
